src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -750,9 +750,9 @@ private static string GetMatchingFrameworkStrings(PackageSpec spec, NuGetFramewo
 
             foreach (var item in items)
             {
-                var dependencies =
-                    item.Data?.Dependencies?.Select(
-                        dependency => new PackageDependency(dependency.Name, VersionRange.All));
+                IEnumerable<PackageDependency> dependencies = item.Data?.Dependencies?
+                    .Where(i => i.ReferenceType != LibraryDependencyReferenceType.None) // Ignore transitively pinned dependencies
```

Comment:
As to why it's happening, is it maybe because the item.Data.Dependencies is likely deterministic as it's what's written in the nuspec.

Reply:
🤷‍♂️